projects/plugins/jetpack/_inc/polldaddy-shortcode.js

```diff
@@ -1,74 +1,94 @@
-( function ( $ ) {
-	window.polldaddyshortcode = {
+( function ( w, d ) {
+	w.polldaddyshortcode = {
 		render: function () {
-			var ratings = $( 'div.pd-rating[data-settings]' );
-			var polls = $( 'div.PDS_Poll[data-settings]' );
+			const ratings = d.querySelectorAll( 'div.pd-rating[data-settings]' );
+			const polls = d.querySelectorAll( 'div.PDS_Poll[data-settings]' );
 
-			if ( polls ) {
-				$.each( polls, function () {
-					var poll = $( this ).data( 'settings' );
+			polls.forEach( pollEl => {
+				const isInitialized = pollEl.getAttribute( 'data-pd-init-done' );
 
-					if ( poll ) {
-						var poll_url = document.createElement( 'a' );
-						poll_url.href = poll[ 'url' ];
-						if (
-							poll_url.hostname != 'secure.polldaddy.com' &&
-							poll_url.hostname != 'static.polldaddy.com'
-						) {
-							return false;
-						}
-						var pathname = poll_url.pathname;
-						if ( ! /\/?p\/\d+\.js/.test( pathname ) ) {
-							return false;
-						}
-						var wp_pd_js = document.createElement( 'script' );
-						wp_pd_js.type = 'text/javascript';
-						wp_pd_js.src = poll[ 'url' ];
-						wp_pd_js.charset = 'utf-8';
-						wp_pd_js.async = true;
-						document.getElementsByTagName( 'head' )[ 0 ].appendChild( wp_pd_js );
+				if ( isInitialized ) {
+					return;
+				}
+
+				pollEl.setAttribute( 'data-pd-init-done', '1' );
+				const settings = pollEl.getAttribute( 'data-settings' );
+				const poll = settings ? JSON.parse( settings ) : null;
```

Comment:
Should we handle any errors that might be thrown if the settings are not valid JSON?

(jQuery's `.data()` returns the value as a string if it doesn't parse as JSON)

Reply:
Yup, that makes sense! Created a little wrapper function.